src/components/CsvUpload/DownloadCsvTemplateButton.tsx

```diff
@@ -0,0 +1,52 @@
+import { type PropsWithChildren } from 'react'
+import { Button } from '../Button'
+import { FileDownIcon } from 'lucide-react'
+import { format } from 'date-fns'
+
+interface DownloadCsvTemplateButtonProps<T> {
+  csvProps: {
+    headers: (keyof T)[]
+    rows?: T[]
+  }
+  fileName?: string
+  className?: string
+}
+
+function isDate(val: unknown): val is Date {
+  return val instanceof Date
+}
+
+export const DownloadCsvTemplateButton = <T extends Record<string, any>>({ children, className, csvProps, fileName = 'template.csv' }: PropsWithChildren<DownloadCsvTemplateButtonProps<T>>) => {
+  const { headers, rows = [] } = csvProps
+  const handleDownload = () => {
+    const csvData: string[][] = [
+      headers as string[],
+      ...rows.map((row: T) =>
+        headers.map((header: keyof T) => {
+          const value = row[header]
+          if (isDate(value)) return format(value, 'MM/dd/yyyy')
+          if (typeof value === 'number') return value.toString()
+          return value ?? ''
+        }),
+      ),
+    ]
+    const csvContent = csvData.map(row => row.map(value => `"${value.replace(/"/g, '""')}"`).join(',')).join('\n')
+
+    const blob = new Blob([csvContent], { type: 'text/csv;charset=utf-8' })
+    const url = URL.createObjectURL(blob)
+
+    const link = document.createElement('a')
```

Comment:
Prefer setting a `display: none` before adding these to prevent flashing, though that's purely defensive I didn't see anything in practice.